pkgs/applications/misc/mediainfo-gui/default.nix

```diff
@@ -1,28 +1,28 @@
-{ stdenv, fetchurl, automake, autoconf, libtool, pkgconfig, libzen, libmediainfo, wxGTK, desktop_file_utils, libSM, imagemagick }:
+{ stdenv, fetchurl, autoreconfHook, pkgconfig, libzen, libmediainfo, wxGTK
+, desktop_file_utils, libSM, imagemagick }:
 
 stdenv.mkDerivation rec {
-  version = "0.7.82";
+  version = "0.7.83";
   name = "mediainfo-gui-${version}";
   src = fetchurl {
     url = "http://mediaarea.net/download/source/mediainfo/${version}/mediainfo_${version}.tar.xz";
-    sha256 = "0ivvmxx93aldfbms6wg46x9npghg304j2zxl5i70m710gybjr232";
+    sha256 = "0d8mph9lbg2lw0ccg1la0kqhbisra8q9rzn195lncch5cia5zyg7";
   };
 
-  buildInputs = [ automake autoconf libtool pkgconfig libzen libmediainfo wxGTK desktop_file_utils libSM imagemagick ];
+  nativeBuildInputs = [ autoreconfHook pkgconfig libzen libmediainfo wxGTK
+                        desktop_file_utils libSM imagemagick ];
```

Comment:
Oh. I think you misunderstood me here. `autoreconfHook` `pkgconfig` are `nativeBuildInputs` the remaining are `buildInputs`. The difference isn't document well, here's a example:
Assume we are cross building an application for ARM on a x86 host, this means we need the following:
`gcc` is needed on the host machine -> x86
`pkgconfig`, `make`, `autoconf` are run on the host as well and never needed on the target machine -> x86
`zlib` `wxGtk`.. are build & runtime dependencies for the app, these are needed for ARM architecture and are never run on the host

To tell our build system whats needed in which architecture we place build tools in `nativeBuildInputs` (= native to the building machine) and application dependencies in `buildInputs` (= native to target architecture).